src/diff/children.js

```diff
@@ -310,7 +310,12 @@ function placeChild(
 	if (nextDom !== undefined) {
 		oldDom = nextDom;
 	} else {
-		oldDom = newDom.nextSibling;
+		oldDom = newDom;
+		// Skip over comment nodes. Will be removed right after calling diffChildren
+		// in diff so that the vnode tree matches the DOM tree again.
+		do {
+			oldDom = oldDom.nextSibling;
+		} while (oldDom !== null && oldDom.nodeType === 8);
```

Comment:
I wonder if it would be possible to use a similar (or the same somehow?) check to the one we have for node types here?

https://github.com/preactjs/preact/blob/de08e919860de81ec5f1ee7a37d640eb1496c323/src/diff/index.js#L358-L359

The other one knows what type it's looking for, so maybe it isn't possible? Just trying to think of ways to avoid checking the `nodeType` getter.